.pre-commit-config.yaml

```diff
@@ -16,7 +18,7 @@ repos:
         - id: ruff
           args: [ --config=pyproject.toml ]
     - repo: https://github.com/psf/black
-      rev: 23.1.0
+      rev: 22.12.0
```

Comment:
Is there a reason this is being downgraded?

Reply:
No, don't know how the downgrade crept it. I'll revert it to 23.1.0

Reply:
I didn't change any settings, so I assume (?) that anything that's going wrong here may also go wrong for users. I don't have any branch protection rules set up. The `GITHUB_TOKEN` permissions are set to `Workflows have read and write permissions in the repository for all scopes.`

Reply:
Same happens on a new, public repo. May be something with my settings somewhere? Could someone else (@neuroinformatics-unit/neuroinformatics-team) try following the instructions to create a repo and host docs?

Reply:
I am trying to do that myself right now. It's the curse of this PR, never-ending 😉

Reply:
I get [the same error](https://github.com/niksirbi/cookiecutter-test-repo/actions/runs/5060026735/jobs/9082387671)

Reply:
I think I tracked down the source of this error.
There was [some helpful information](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token) in the GitHbu action I am using for publishing on gh-pages.
Turns out that we have to enable [write permissions for the `GITHUB-TOKEN`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) in the workflow or specific job.

@adamltyson try modifying the `deploy_sphinx_docs` job in your test repo, thus:
```yaml
deploy_sphinx_docs:
    name: Deploy Sphinx Docs
    needs: build_sphinx_docs
    permissions:
      contents: write
    if: github.event_name == 'push' && github.ref_type == 'tag'
    runs-on: ubuntu-latest
    steps:
      - uses: neuroinformatics-unit/actions/deploy_sphinx_docs@v2
        with:
          secret_input: ${{ secrets.GITHUB_TOKEN }}
```
This worked for me (see [case in point](https://niksirbi.github.io/cookiecutter-test-repo/)). If it works for you too, I'll change it here and document it.

Reply:
Yep works for me :tada: